src/ILVerification/src/ILVerifyTypeSystemContext.cs

```diff
@@ -47,6 +47,7 @@ internal override ModuleDesc ResolveModule(ModuleDesc referencingModule, string
             {
                 throw new VerifierException($"The module '{fileName}' is not expected to be an assembly");
             }
+            module.ReferencingModule = referencingModule;
```

Comment:
Check that it is null before setting it, and print error if it is non-null.

Technically, it is ok for one module to be referenced by multiple assemblies, but I do not think it is ever used in practice. It is ok to not support this corner case in IL verify.